res/controllers/Hercules-DJControl-Inpulse-300-script.js

```diff
@@ -0,0 +1,234 @@
+// * Mixxx mapping script file for the Hercules DJControl Inpulse 300.
+// * Author: DJ Phatso, contributions by Kerrick Staley
+// * Version 1.2 (December 2019)
+// * Forum: https://www.mixxx.org/forums/viewtopic.php?f=7&t=12599
+// * Wiki: https://mixxx.org/wiki/doku.php/hercules_djcontrol_inpulse_300
+//
+// Changes to v1.2
+// - Code cleanup.
+//
+// Changes to v1.1
+// - Fix seek-to-start and cue-master behavior.
+// - Tweak scratch, seek, and bend behavior.
+// - Controller knob/slider values are queried on startup, so MIXXX is synced.
+// - Fixed vinyl button behavior the first time it's pressed.
+//
+// v1.0 : Original forum release
+//
+// TODO: Functions that could be implemented to the script:
+//
+// * ROLL: Keep SLIP active (if already enabled) when exiting from rolls
+//
+// * SLICER/SLICER LOOP
+//
+// * TONEPLAY
+//
+// * FX:
+//	- Potentially use 1 FX rack for FX pads and another for the Controlled FX
+//  - See how to preselect effects for a rack
+// * Fix behavior when adjusting tempo slider after pressing [Sync] (tempo adjustment should be relative, not absolute).
+// ****************************************************************************
+
+DJCi300.kScratchActionNone = 0;
+DJCi300.kScratchActionScratch = 1;
+DJCi300.kScratchActionSeek = 2;
+DJCi300.kScratchActionBend = 3;
+
+//function DJCi300() {}
+
+DJCi300.vuMeterUpdate = function(value, group, control) {
+    value = (value * 127) + 5;
+    switch (control) {
+    case "VuMeterL":
+        midi.sendShortMsg(0xB0, 0x40, value);
+        break;
+    case "VuMeterR":
+        midi.sendShortMsg(0xB0, 0x41, value);
+        break;
+    }
+};
+
+DJCi300.vuMeterUpdateDA = function(value, group, control) {
+    value = (value * 127) + 5;
+    switch (control) {
+    case "[Channel1]", "VuMeter":
+        midi.sendShortMsg(0xB1, 0x40, value);
+        break;
+    }
+};
+
+DJCi300.vuMeterUpdateDB = function(value, group, control) {
+    value = (value * 127) + 5;
+    switch (control) {
+    case "[Channel2]", "VuMeter":
+        midi.sendShortMsg(0xB2, 0x40, value);
+        break;
+    }
+};
+
+DJCi300.init = function() {
+    DJCi300.scratchButtonState = true;
+    DJCi300.scratchAction = {
+        1: DJCi300.kScratchActionNone,
+        2: DJCi300.kScratchActionNone
+    };
+
+    // Turn On Vinyl buttons LED(one for each deck).
+    midi.sendShortMsg(0x91, 0x03, 0x7F);
+    midi.sendShortMsg(0x92, 0x03, 0x7F);
+
+    //Turn On Browser button LED
+    midi.sendShortMsg(0x90, 0x05, 0x10);
+
+    // Connect the VUMeters
+    engine.connectControl("[Channel1]", "VuMeter", "DJCi300.vuMeterUpdateDA");
+    engine.connectControl("[Channel2]", "VuMeter", "DJCi300.vuMeterUpdateDB");
+    engine.connectControl("[Master]", "VuMeterL", "DJCi300.vuMeterUpdate");
+    engine.connectControl("[Master]", "VuMeterR", "DJCi300.vuMeterUpdate");
+
+    // Connect the Browser LEDs
+    engine.getValue("[Library]", "MoveFocus");
+    engine.getValue("[Master]", "maximize_library");
```

Comment:
Why do you read these values if you don't use them?

Reply:
I figured it probably wasn't working since it didn't do what I wanted, but it did help with lighting up the LED around the Browser knob.

My goal was to vary the color when the focus is changed between the two main part of the browser (Ex: Green when on Folder/ Blue for files).

Reply:
Please try to remove these two `engine.getValue()` calls and check if it still works. I don't think they have anything to do with the LEDs.